src/runtime/tls_openssl.rs

```diff
@@ -66,25 +71,43 @@ pub(super) async fn tls_connect(
     Ok(stream)
 }
 
-fn make_openssl_connector(cfg: TlsOptions) -> std::result::Result<SslConnector, ErrorStack> {
-    let mut builder = SslConnector::builder(SslMethod::tls_client())?;
+fn make_openssl_connector(cfg: TlsOptions) -> Result<SslConnector> {
+    let openssl_err = |e: ErrorStack| {
+        Error::from(ErrorKind::InvalidTlsConfig {
+            message: e.to_string(),
+        })
+    };
+
+    let mut builder = SslConnector::builder(SslMethod::tls_client()).map_err(openssl_err)?;
 
     let TlsOptions {
         allow_invalid_certificates,
         ca_file_path,
         cert_key_file_path,
         allow_invalid_hostnames: _,
+        tls_certificate_key_file_password,
     } = cfg;
 
     if let Some(true) = allow_invalid_certificates {
         builder.set_verify(SslVerifyMode::NONE);
     }
     if let Some(path) = ca_file_path {
-        builder.set_ca_file(path)?;
+        builder.set_ca_file(path).map_err(openssl_err)?;
     }
     if let Some(path) = cert_key_file_path {
-        builder.set_certificate_file(path.clone(), SslFiletype::PEM)?;
-        builder.set_private_key_file(path, SslFiletype::PEM)?;
+        builder
+            .set_certificate_file(path.clone(), SslFiletype::PEM)
+            .map_err(openssl_err)?;
+        if let Some(key_pw) = tls_certificate_key_file_password {
+            let contents = std::fs::read(&path)?;
+            let key_bytes = decrypt_private_key(&contents, &key_pw)?;
+            let key = openssl::pkey::PKey::private_key_from_pem(&key_bytes).map_err(openssl_err)?;
```

Comment:
```suggestion
            let key = openssl::pkey::PKey::private_key_from_der(&key_bytes).map_err(openssl_err)?;
```

Fixes observed error testing locally:

```
Error: Error { kind: InvalidTlsConfig { message: "error:1E08010C:DECODER routines:OSSL_DECODER_from_bio:unsupported:crypto/encode_decode/decoder_lib.c:102:No supported data to decode. Input type: PEM" }, labels: {}, wire_version: None, source: None }
```

IIUC, PEM is the base64 encoding, DER is the binary encoding.